zfsbackup.py

```diff
@@ -532,6 +535,107 @@ def has_backuplast(dataset, inc_name):
     else:
         return False
 
+def clean_dest_snaps(destinations, global_retain_snaps=None):
+    """
+       delete all but the n snapshots from destinations per config
+       param destinations: list of destinations from config file
+       param global_retain_snaps: number of snapshots that should be kept
+       as defined by the retain_snaps global config param.
+    """
+    for dest in destinations:
+        dataset = dest.get('dest')
+        transport = dest.get('transport')
+        if dest.get('retain_snaps') is None and global_retain_snaps is None:
+            # We're not deleting anything
+            logging.info("Not cleaning up snaps for: "+dataset
+                         + " via " +transport)
+            return
+        elif dest.get('retain_snaps') is None:
+            num_snaps = global_retain_snaps
+        else:
+            num_snaps = dest.get('retain_snaps')
+        zfs_command = ['zfs', 'list', '-H', '-r', '-t', 'snapshot',
+                       '-o', 'name', dataset]
+        if transport.lower() == 'local':
+            # local transport
+            snaps = __snap_delete_format(__run_command(zfs_command), num_snaps)
+            errors = 0
+            logging.info("Deleting "+str(len(snaps))+ " from "
+                         + dataset + " via " +transport)
+            for snap in snaps:
+                try:
+                    delete_snapshot(snap)
+                except ZFSBackupError:
+                    errors += 1
+            if errors > 0:
+                logging.warn("Encountered errors while deleting old snapshots" 
+                             + "from destination: "+dataset+" via "
+                             + transport)
+        elif transport.lower().split(':')[0] == 'ssh':
```

Comment:
the ssh parsing here seems to also be repeated in verify_backup and send_snapshot - maybe breakout to a function?

Reply:
This should be addressed now. There are other areas where code reuse is a thing, but I'd like to save that for a separate cleanup PR.